src/main/java/htsjdk/samtools/util/ReadEndsIterator.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2010 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package htsjdk.samtools.util;
+
+import htsjdk.samtools.AlignmentBlock;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.SamReader;
+
+/**
+ * Iterator that traverses a SAM File, accumulating information on a per-locus basis.
+ * Optionally takes a target interval list, in which case the loci returned are the ones covered by
+ * the interval list.  If no target interval list, whatever loci are covered by the input reads are returned.
+ * By default duplicate reads and non-primary alignments are filtered out.  Filtering may be changed
+ * via setSamFilters(). Difference from SamLocusIterator is that this implementation accumulates data
+ * only about start and end of alignment blocks from reads, not about each aligned base.
+ */
+public class ReadEndsIterator extends AbstractLocusIterator<TypedRecordAndOffset, AbstractLocusInfo<TypedRecordAndOffset>> {
```

Comment:
ReadEnds is not a good name. it's reminiscent of the ReadEnds object in MarkDuplicates, but seems to have nothing to do with it...please come up with a better name...one that has, perhaps `AlignmentBlocks` in it?